Lib/test/test_descr.py

```diff
@@ -1458,6 +1458,31 @@ class someclass(metaclass=dynamicmetaclass):
             pass
         self.assertNotEqual(someclass, object)
 
+    def test_attribute_assignment(self):
+        class A(type):
+            def __setattr__(cls, key, value):
+                type.__setattr__(cls, key, value)
+
+        class B:
+            pass
+
+        class C(B, A):
+            pass
+
+        obj = C('D', (object,), {})
+        try:
+            obj.test = True
+        except TypeError:
+            self.fail("assignment to classes that override __setattr__ should be legal")
+
+        try:
+            str.test = True
+        except TypeError:
+            pass
+        else:
+            self.fail("assignment to bultin classes should be ilegal")
```

Comment:
You don't need to test this. The scenario is already covered by other tests. (And there is a typo, too.)